### PR TITLE
Coupons UI fixes

### PR DESCRIPTION
**Pull request by dsego**
- Remove "Your Platform Name Here" from footer (SOL-1540)
- Update coupon collection url.
- Suggested UI text edits in comment section of SOL-1356
- Total Paid field can be empty SOL-1555
- Fixes issues in adding a course ID SOL-1554

@mjfrey @vkaracic @marjev

### CODE REVIEW

**Comment by clintonb** [comment]
You can use a [number input](http://www.w3schools.com/html/html_form_input_types.asp) so the browser does some validation for you. Compare https://github.com/edx/ecommerce/blob/master/ecommerce/static/templates/coupon_form.html#L57 with https://github.com/edx/ecommerce/blob/master/ecommerce/static/templates/professional_course_seat_form_field.html#L13-L14

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @dsego! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=479

**Comment by dsego** [comment]
Ready for review.

@marjev @vkaracic @clintonb @AlasdairSwan

File: ecommerce/static/js/test/specs/views/coupon_create_view_spec.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ define([
             });
 
             it('should throw an error if submitted with blank fields', function () {
-                var errorHTML = '<strong>Error!</strong> Please complete all required fields.';
+                var errorHTML = '<strong></strong> You must complete all required fields.';
```
Comment (vkaracic) [inline ecommerce/static/js/test/specs/views/coupon_create_view_spec.js:52]: Do you still need the `<strong>` tag here?

Comment (dsego) [inline ecommerce/static/js/test/specs/views/coupon_create_view_spec.js:52]: The strong tag gets rendered even if alert title is missing. It keeps the alert template logic-less.

**Comment by vkaracic** [comment]
LGTM :+1:

**Comment by marjev** [comment]
:+1:

File: ecommerce/static/js/collections/coupon_collection.js [hunk 2]
```diff
@@ -14,14 +14,7 @@ define([
...
             model: CouponModel,
-            url: '/api/v2/products/',
-            fetch: function (options) {
-                options = options || {};
-                _.defaults(options.data || (options.data = {}), {
-                    product_class__name: 'Coupon'
-                });
-                return this._super(options);
-            }
+            url: '/api/v2/coupons/'
```
Comment (clintonb) [inline ecommerce/static/js/collections/coupon_collection.js:17]: This API endpoint only supports POST operations.

File: ecommerce/static/js/models/coupon_model.js [hunk 3]
```diff
@@ -18,6 +18,11 @@ define([
             ) {
         'use strict';
 
+        _.extend(Backbone.Validation.messages, {
+            required: gettext('This field is required'),
+            number: gettext('This value must be a number'),
```
Comment (clintonb) [inline ecommerce/static/js/models/coupon_model.js:23]: Merge with digits. No need for both.

File: ecommerce/static/js/models/coupon_model.js [hunk 4]
```diff
@@ -26,47 +31,36 @@ define([
...
                     var startDate,
                         endDate;
                     if (_.isEmpty(val)) {
-                        return gettext('Start date is required');
+                        return Backbone.Validation.messages.required;
                     }
                     startDate = moment(new Date(val));
                     if (!startDate.isValid()) {
-                        return gettext('Start date is invalid');
+                        return gettext('This value must be a date');
```
Comment (clintonb) [inline ecommerce/static/js/models/coupon_model.js:63]: This message is duplicated.

File: ecommerce/static/js/test/specs/views/coupon_create_view_spec.js (hunk 1 above)
Comment (clintonb) [inline ecommerce/static/js/test/specs/views/coupon_create_view_spec.js:52]: I don't follow. Why is there an empty tag here? I am missing how this is logic-less.

**Comment by clintonb** [comment]
You have six separate changes in a single PR. I need you to do the following:
1. File a separate PR for each change. Tedious? Yes. But it will ensure reviewers don't gloss over anything.
2. Add screenshots for UI changes.

**Comment by dsego** [comment]
I will propose separate PRs.

**Pull request closed without merging**